File: docs/breez-sdk/snippets/flutter/lib/receive_payment.dart
import 'package:breez_sdk_spark_flutter/breez_sdk_spark.dart';

Future<ReceivePaymentResponse> receivePaymentLightning(
    BreezSdk sdk) async {
  // ANCHOR: receive-payment-lightning-bolt11
  String description = "<invoice description>";
  // Optionally set the invoice amount you wish the payer to send
  BigInt optionalAmountSats = BigInt.from(5000);
  // Optionally set the expiry duration in seconds
  int optionalExpirySecs = 3600;

  // Create an invoice and set the amount you wish the payer to send
  ReceivePaymentRequest request = ReceivePaymentRequest(
      paymentMethod: ReceivePaymentMethod.bolt11Invoice(
          description: description,
          amountSats: optionalAmountSats,
          expirySecs: optionalExpirySecs,
          paymentHash: null));
  ReceivePaymentResponse response = await sdk.receivePayment(
    request: request,
  );

  String paymentRequest = response.paymentRequest;
  print("Payment request: $paymentRequest");
  BigInt receiveFeeSats = response.fee;
  print("Fees: $receiveFeeSats sats");
  // ANCHOR_END: receive-payment-lightning-bolt11
  return response;
}

Future<ReceivePaymentResponse> receivePaymentOnchain(
    BreezSdk sdk) async {
  // ANCHOR: receive-payment-onchain
  ReceivePaymentRequest request = ReceivePaymentRequest(
      paymentMethod: ReceivePaymentMethod.bitcoinAddress());
  ReceivePaymentResponse response = await sdk.receivePayment(
    request: request,
  );

  String paymentRequest = response.paymentRequest;
  print("Payment request: $paymentRequest");
  BigInt receiveFeeSats = response.fee;
  print("Fees: $receiveFeeSats sats");
  // ANCHOR_END: receive-payment-onchain
  return response;
}

Future<ReceivePaymentResponse> receivePaymentSparkAddress(BreezSdk sdk) async {
  // ANCHOR: receive-payment-spark-address
  ReceivePaymentRequest request =
      ReceivePaymentRequest(paymentMethod: ReceivePaymentMethod.sparkAddress());
  ReceivePaymentResponse response = await sdk.receivePayment(
    request: request,
  );

  String paymentRequest = response.paymentRequest;
  print("Payment request: $paymentRequest");
  BigInt receiveFeeSats = response.fee;
  print("Fees: $receiveFeeSats sats");
  // ANCHOR_END: receive-payment-spark-address
  return response;
}

Future<ReceivePaymentResponse> receivePaymentSparkInvoice(BreezSdk sdk) async {
  // ANCHOR: receive-payment-spark-invoice
  String optionalDescription = "<invoice description>";
  BigInt optionalAmountSats = BigInt.from(5000);
  // Optionally set the expiry UNIX timestamp in seconds
  BigInt optionalExpiryTimeSeconds = BigInt.from(1716691200);
  String optionalSenderPublicKey = "<sender public key>";

  ReceivePaymentRequest request =
      ReceivePaymentRequest(paymentMethod: ReceivePaymentMethod.sparkInvoice(
        description: optionalDescription,
        amount: optionalAmountSats,
        expiryTime: optionalExpiryTimeSeconds,
        senderPublicKey: optionalSenderPublicKey,
      ));
  ReceivePaymentResponse response = await sdk.receivePayment(
    request: request,
  );

  String paymentRequest = response.paymentRequest;
  print("Payment request: $paymentRequest");
  BigInt receiveFeeSats = response.fee;
  print("Fees: $receiveFeeSats sats");
  // ANCHOR_END: receive-payment-spark-invoice
  return response;
}

File: docs/breez-sdk/snippets/react-native/receive_payment.ts
import {
  ReceivePaymentMethod,
  type BreezSdk
} from '@breeztech/breez-sdk-spark-react-native'

const exampleReceiveLightningPayment = async (sdk: BreezSdk) => {
  // ANCHOR: receive-payment-lightning-bolt11
  const description = '<invoice description>'
  // Optionally set the invoice amount you wish the payer to send
  const optionalAmountSats = BigInt(5_000)
  // Optionally set the expiry duration in seconds
  const optionalExpirySecs = 3600

  const response = await sdk.receivePayment({
    paymentMethod: new ReceivePaymentMethod.Bolt11Invoice({
      description,
      amountSats: optionalAmountSats,
      expirySecs: optionalExpirySecs,
      paymentHash: undefined
    })
  })

  const paymentRequest = response.paymentRequest
  console.log(`Payment Request: ${paymentRequest}`)
  const receiveFeeSats = response.fee
  console.log(`Fees: ${receiveFeeSats} sats`)
  // ANCHOR_END: receive-payment-lightning-bolt11
}

const exampleReceiveOnchainPayment = async (sdk: BreezSdk) => {
  // ANCHOR: receive-payment-onchain
  const response = await sdk.receivePayment({
    paymentMethod: new ReceivePaymentMethod.BitcoinAddress()
  })

  const paymentRequest = response.paymentRequest
  console.log(`Payment Request: ${paymentRequest}`)
  const receiveFeeSats = response.fee
  console.log(`Fees: ${receiveFeeSats} sats`)
  // ANCHOR_END: receive-payment-onchain
}

const exampleReceiveSparkAddress = async (sdk: BreezSdk) => {
  // ANCHOR: receive-payment-spark-address
  const response = await sdk.receivePayment({
    paymentMethod: new ReceivePaymentMethod.SparkAddress()
  })

  const paymentRequest = response.paymentRequest
  console.log(`Payment Request: ${paymentRequest}`)
  const receiveFeeSats = response.fee
  console.log(`Fees: ${receiveFeeSats} sats`)
  // ANCHOR_END: receive-payment-spark-address
}

const exampleReceiveSparkInvoice = async (sdk: BreezSdk) => {
  // ANCHOR: receive-payment-spark-invoice
  const optionalDescription = '<invoice description>'
  const optionalAmountSats = BigInt(5_000)
  // Optionally set the expiry UNIX timestamp in seconds
  const optionalExpiryTimeSeconds = BigInt(1716691200)
  const optionalSenderPublicKey = '<sender public key>'

  const response = await sdk.receivePayment({
    paymentMethod: new ReceivePaymentMethod.SparkInvoice({
      description: optionalDescription,
      amount: optionalAmountSats,
      expiryTime: optionalExpiryTimeSeconds,
      senderPublicKey: optionalSenderPublicKey,
      tokenIdentifier: undefined
    })
  })

  const paymentRequest = response.paymentRequest
  console.log(`Payment Request: ${paymentRequest}`)
  const receiveFeeSats = response.fee
  console.log(`Fees: ${receiveFeeSats} sats`)
  // ANCHOR_END: receive-payment-spark-invoice
}
